classes/User.php

```diff
@@ -958,6 +968,13 @@ public function checkBan($type):bool {
         return false;
     }
 
+    public function loadTrainingManager(): void
+    {
+        $this->trainingManager = new TrainingManager($this->system, $this->train_type, $this->train_gain,
+    $this->train_time, $this->rank, $this->forbidden_seal, $this->reputation, $this->team, $this->sensei_id,
+            $this->bloodline_id);
+    }
+
```

Comment:
Relying on this method being called for `trainingManager` to be present introduces a lot of bug possibilities where you call `$player->trainingManager` but it wasn't loaded. We could fix this in a couple ways:

1) Always instantiate this when the user class is loaded (call this method in `loadData()`)
2) Make the property private and make this method `getTrainingManager()`. Inside this we can return `$this->trainingManager` if already initialized, if not we initialize first.